src/client.rs

```diff
@@ -91,9 +119,12 @@ impl<Req, Resp, E> fmt::Debug for Client<Req, Resp, E>
 }
 
 /// Additional options to configure how the client connects and operates.
-#[derive(Clone, Default)]
+#[cfg_attr(feature = "tls", derive(Default))]
+#[cfg_attr(not(feature = "tls"), derive(Clone, Default))]
```

Comment:
Okay, yeah I wasn't sure if you had some need for `Clone` but I agree it's better if it doesn't change between features.